tests/impl.cpp

```diff
@@ -723,6 +726,17 @@ bool validateSingleFloatPair(float a, float b)
                          // into account as well.
 }
 
+bool validateSingleDoublePair(double a, double b)
```

Comment:
I am not sure if this function is worthy as a standalone helper function.

Reply:
As in the future we might integrate more intrinsics related to double value validation?

Reply:
`validateDouble` can be generic enough to satisfy validation needs.

Reply:
I copied the original design of `validateFloat()` and `validateSingleFloatPair()`.
Maybe they should be integrated into a single function?
Like integrate `validateFloat()` and `validateSingleFloatPair()` into `validateFloat()`?